tests/test_comparison_level_composition.py

```diff
@@ -72,16 +52,47 @@ def test_binary_composition_internals(clause, c_fun):
         m_probability=0.5,
     )
 
-    exact_match_sql = (
-        f'("first_name_l" = "first_name_r") {clause} ("surname_l" = "surname_r")'
-    )
+    exact_match_sql = f"({q}first_name_l{q} = {q}first_name_r{q}) {clause} ({q}surname_l{q} = {q}surname_r{q})"  # noqa: E501
     assert level.sql_condition == f"NOT ({exact_match_sql})"
 
     with pytest.raises(ValueError):
         c_fun()
 
 
-def test_composition_outputs():
+@mark_with_dialects_excluding("postgres")
+def test_binary_composition_internals_OR(test_helpers, dialect):
+    cll = test_helpers[dialect].cll
+    quo, _ = _get_dialect_quotes(dialect)
+    binary_composition_internals("OR", cll.or_, cll, quo)
+
+
+@mark_with_dialects_excluding("postgres")
+def test_binary_composition_internals_AND(test_helpers, dialect):
+    cll = test_helpers[dialect].cll
+    quo, _ = _get_dialect_quotes(dialect)
+    binary_composition_internals("AND", cll.and_, cll, quo)
+
+
+@mark_with_dialects_including("duckdb")
```

Comment:
I wonder if it makes more sense to leave this unmarked - we aren't really testing duckdb itself, but just using it as a tool to check the universal `cll.not_`, and I think we can expect anyone running the tests to have `duckdb` installed as a core requirement. Not a big deal though so happy if you disagree